src/dash-table/components/Table/index.tsx

```diff
@@ -49,24 +50,45 @@ export default class Table extends Component<SanitizedAndDerivedProps, Standalon
     }
 
     componentWillReceiveProps(nextProps: SanitizedAndDerivedProps) {
-        if (nextProps.filter_query === this.props.filter_query) {
-            return;
-        }
-
         this.setState(state => {
-            const { workFilter: { map: currentMap, value } } = state;
-
-            if (value !== nextProps.filter_query) {
-                const map = this.filterMap(
-                    currentMap,
-                    nextProps.filter_query,
-                    nextProps.visibleColumns
-                );
-
-                return map !== currentMap ? { workFilter: { map, value} } : null;
-            } else {
-                return null;
+            const { applyFocus: currentApplyFocus, workFilter: { map: currentMap, value } } = state;
+
+            const nextState: Partial<StandaloneState> = {};
+
+            // state for filter
+            if (nextProps.filter_query !== this.props.filter_query) {
+                if (value !== nextProps.filter_query) {
+                    const map = this.filterMap(
+                        currentMap,
+                        nextProps.filter_query,
+                        nextProps.visibleColumns
+                    );
+
+                    if (map !== currentMap) {
+                        nextState.workFilter = { map, value };
+                    }
+                }
+            }
+
+            // state for applying focus
+            if (nextProps.active_cell !== this.props.active_cell) {
+                if (!currentApplyFocus) {
+                    nextState.applyFocus = true;
+                }
+            } else if (nextProps.loading_state !== this.props.loading_state) {
+                let applyFocus = false;
+
+                const activeElement = document.activeElement as HTMLElement;
+                if (activeElement) {
+                    applyFocus = !!DOM.getParentById(activeElement, this.props.id);
+                }
+
+                if (applyFocus !== currentApplyFocus) {
+                    nextState.applyFocus = applyFocus;
+                }
```

Comment:
State will be updated when next render is run.

Only foce refocus when (1) the active_cell has changed, (2) the focus is currently inside the table, otherwise this means the focus is elsewhere and there's either no point forcing focus or it will cause usability issues..